serde/src/de/value.rs

```diff
@@ -22,31 +23,39 @@ use bytes;
 #[derive(Clone, Debug, PartialEq)]
 pub enum Error {
     /// The value had some syntatic error.
-    SyntaxError,
+    Syntax(String),
+
+    /// The value had an incorrect type.
+    Type(de::Type),
+
+    /// The value had an invalid length.
+    Length(usize),
```

Comment:
There are three possible solutions that I see here (the last option being my preferred):
- Remove `value::Error` because `ValueDeserializer`s are assumed to be used in a nested manner.
- Remove just the `Length` variant because existing implementations don't use it.
- Keep the status quo and support all possibilities, for example other implementations which may use `de::Error::length_mismatch`.

What do you think is sensible?